File: src/main/scala/com/vc/medium/L215.scala
package com.vc.medium

object L215 {
  def findKthLargest(arr: Array[Int], k: Int): Int = {
    import scala.collection.mutable
    val pq = new mutable.PriorityQueue[Int]()(Ordering[Int].reverse)

    var i = 0
    val n = arr.length
    while(i < n) {
      pq.enqueue(arr(i))
      if(pq.size > k) pq.dequeue
      i += 1
    }
    pq.head
  }
}

File: src/main/java/com/vc/medium/L215.java
package com.vc.medium;

import java.util.PriorityQueue;

class L215 {
    public int findKthLargest(int[] arr, int k) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for(int i = 0; i < arr.length; i++) {
            pq.offer(arr[i]);
            if(pq.size() > k) pq.poll();
        }
        return pq.isEmpty() ? -1: pq.poll();
    }
}